packages/drivers/driver-web-cache/src/FluidCache.ts

```diff
@@ -117,10 +120,35 @@ export class FluidCache implements IPersistedCache {
 		});
 	}
 
+	private async openDb() {
+		if (this.db === undefined) {
+			this.db = await getFluidCacheIndexedDbInstance(this.logger);
+			// Need to close the db on version change if opened.
+			this.db.addEventListener("versionchange", (ev) => {
+				this.db?.close();
+				this.db = undefined;
+			});
```

Comment:
Sorry, I did miss this change with my earlier comments. I still think it would be good to guarantee this would apply to all cases with unit tests/patterns/etc. if there is no guarantee of that already.

Also good to note this will cause all existing DB sessions to throw errors, which could impact FF and the hosting app. Do you know what FF and app behavior will be in these cases?